### PR TITLE
Update Currently Inked to new layout features + refactor

### PR DESCRIPTION
**Pull request by wkillerud**
Hia Urban, hope all is well. Sending some suggested changes your way, to bring feature parity to Currently inked. 

Sorry for the large diff 😅 Commit by commit might help for review.

### Changes 
Mirror the feature set of the other views. Move the cards to React, for live filtering and decoupling the layout from the viewport width.

Refactor some duplicated logic to shared hooks:

- Field switch logic
- Hidden fields state
- Layout state (card vs table)

Fix excess gap between cards, and a stray `<td>` in the table footer for currently inked.

### CODE REVIEW

File: app/javascript/src/useHiddenFields.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { useCallback, useEffect, useState } from "react";
+import * as storage from "./localStorage";
+
+export const useHiddenFields = (storageKey, defaultHiddenFields) => {
+  const [hiddenFields, setHiddenFields] = useState(defaultHiddenFields || []);
```
Comment (wkillerud) [inline app/javascript/src/useHiddenFields.js:5]: Here I started with setting the default value on the function parameter (`(storageKey, defaultHiddenFields = [])`), but that caused a render loop. Likely the array is new each time, so it breaks `===`.

**Review by ujh — APPROVED** [review]
I wanted to do that for ages now and couldn't be motivated. Thank you! 💚 

PS: If you run out of work, you could also include the archive page along with it at a later stage. 😉  I believe this is how pens and inks work nowadays. We could get rid of quite a few Ruby code that way.

**Comment by ujh** [comment]
I think you should be able to get rid of this line: https://github.com/ujh/fountainpencompanion/blob/c68277443002b61793e5f29c53159224f8002b75/app/controllers/currently_inked_controller.rb#L3

And you should be able to remove the `:index` from the array in this line: https://github.com/ujh/fountainpencompanion/blob/c68277443002b61793e5f29c53159224f8002b75/app/controllers/currently_inked_controller.rb#L5

Alternatively, this is something I can do later.

**Comment by wkillerud** [comment]
> PS: If you run out of work, you could also include the archive page along with it at a later stage. 😉 I believe this is how pens and inks work nowadays. We could get rid of quite a few Ruby code that way.

Only the inks, for now. If the new pen and currently inked API works the same as for inks, it shouldn't be a problem 👍 But yeah, at a later stage 😄

**Comment by ujh** [comment]
Cool, cool. I am sure my motivation will kick in eventually and I can look at those, too.

**Pull request merged**